### PR TITLE
Upgrade Ruma

### PR DESCRIPTION
**Pull request by jplatte**
… to pull in https://github.com/ruma/ruma/pull/1387.

### CODE REVIEW

File: crates/matrix-sdk-appservice/src/lib.rs [hunk 1]
```diff
@@ -347,18 +341,14 @@ impl AppService {
     /// # Returns
     /// This function may return a UIAA response, which should be checked for
     /// with [`Error::as_uiaa_response()`].
-    pub async fn register_user<'a>(
-        &self,
-        localpart: &'a str,
-        device_id: Option<&'a DeviceId>,
-    ) -> Result<()> {
+    pub async fn register_user(&self, localpart: &str, device_id: Option<&DeviceId>) -> Result<()> {
```
Comment (gnunicorn) [inline crates/matrix-sdk-appservice/src/lib.rs:344]: may I suggest we change our API then, too? The str is pretty awkward to work with if you have e.g. have an `Option<String>` coming in and we have to `to_owned()` them now anyways ...

File: crates/matrix-sdk-appservice/src/lib.rs [hunk 2]
```diff
@@ -347,18 +341,14 @@ impl AppService {
...
         if self.is_user_registered(localpart).await? {
             return Ok(());
         }
         let request = assign!(register::v3::Request::new(), {
-            username: Some(localpart),
-            login_type: Some(&register::LoginType::ApplicationService),
-            device_id,
+            username: Some(localpart.to_owned()),
+            login_type: Some(register::LoginType::ApplicationService),
+            device_id: device_id.map(ToOwned::to_owned),
```
Comment (gnunicorn) [inline crates/matrix-sdk-appservice/src/lib.rs:351]: may I suggest we change our API then, too? The str is pretty awkward to work with if you have e.g. have an Option<String> coming in and we have to to_owned() them now anyways ...

**Review by gnunicorn — APPROVED** [review]
I am a little torn. This adds a bunch of `.to_owned()` now for `&str` given in, but for `data: &[u8]` the input parameter was changed to `data: Vec<u8>`. I feel that we might want to change these str into String (and other types) in our external APIs.... the `&str` isn't particularly nice to work with in the first place, once you are beyond hard-coded values as they are e.g. `Option<String>` coming from clap, which we then re-allocate again with this change...

**Comment by jplatte** [comment]
I was also not entirely sure what to do with `&str` / `Option<&str>` parameters. The upload `&[u8]` to `Vec<u8>` was done in this PR because those payloads are possibly rather big and expensive to clone, but for everything else I tried to be conservative and not change signatures right away. Makes sense?

**Pull request merged**